beacon-chain/blockchain/forkchoice/process_attestation.go

```diff
@@ -93,15 +96,30 @@ func (s *Store) OnAttestation(ctx context.Context, a *ethpb.Attestation) (uint64
 		return 0, err
 	}
 
-	// Use the target state to to validate attestation and calculate the committees.
-	indexedAtt, err := s.verifyAttestation(ctx, baseState, a)
-	if err != nil {
-		return 0, err
-	}
+	// Process aggregated attestation in the queue every `slot/2` duration,
+	// this allows the incoming attestation to aggregate and avoid
+	// process individual attestation.
+	if uint64(time.Now().Unix())%params.BeaconConfig().SecondsPerSlot/2 == 0 {
+		s.attsQueueLock.Lock()
+		defer s.attsQueueLock.Unlock()
```

Comment:
I think it is a good practice to only use `defer` when it is not in any nested statements. Defer runs the line of code after the function ends, but in this usage the lock is only needed for the code inside of the if statement. 

Any following code added to this function may face issues with the lock due to it being locked until the end of the function, rather than until the end of the if statement when the lock is no longer needed.

Reply:
Yea i agree with that